compiler/dialects/snax_stream.py

```diff
@@ -0,0 +1,132 @@
+
+    upper_bounds: ParameterDef[ArrayAttr[IntAttr]]
+    temporal_strides: ParameterDef[ArrayAttr[IntAttr]]
+    spatial_strides: ParameterDef[ArrayAttr[IntAttr]]
+
+    def __init__(
+        self,
+        upper_bounds: ParameterDef[ArrayAttr[IntAttr]] | Sequence[int],
+        temporal_strides: ParameterDef[ArrayAttr[IntAttr]] | Sequence[int],
+        spatial_strides: ParameterDef[ArrayAttr[IntAttr]] | Sequence[int],
+    ):
+        parameters: Sequence[Attribute] = []
+        for arg in (upper_bounds, temporal_strides, spatial_strides):
+            if not isinstance(arg, ArrayAttr):
+                arg = ArrayAttr([IntAttr(x) for x in arg])
+            parameters.append(arg)
+        super().__init__(parameters)
+
+    def print_parameters(self, printer: Printer) -> None:
+        with printer.in_angle_brackets():
+            printer.print_string("ub = [")
+            printer.print_list(self.upper_bounds, lambda attr: printer.print(attr.data))
+            printer.print_string("], ts = [")
+            printer.print_list(
+                self.temporal_strides, lambda attr: printer.print(attr.data)
+            )
+            printer.print_string("], ss = [")
+            printer.print_list(
+                self.spatial_strides, lambda attr: printer.print(attr.data)
+            )
+            printer.print_string("]")
+
+    @classmethod
+    def parse_parameters(cls, parser: AttrParser) -> Sequence[Attribute]:
+        with parser.in_angle_brackets():
+            parser.parse_identifier("ub")
+            parser.parse_punctuation("=")
+            ub = ArrayAttr(
+                IntAttr(i)
+                for i in parser.parse_comma_separated_list(
+                    parser.Delimiter.SQUARE, parser.parse_integer
+                )
+            )
+            parser.parse_punctuation(",")
+            parser.parse_identifier("ts")
+            parser.parse_punctuation("=")
+            ts = ArrayAttr(
+                IntAttr(i)
+                for i in parser.parse_comma_separated_list(
+                    parser.Delimiter.SQUARE, parser.parse_integer
+                )
+            )
+            parser.parse_punctuation(",")
+            parser.parse_identifier("ss")
+            parser.parse_punctuation("=")
+            ss = ArrayAttr(
+                IntAttr(i)
+                for i in parser.parse_comma_separated_list(
+                    parser.Delimiter.SQUARE, parser.parse_integer
+                )
+            )
+            return (ub, ts, ss)
+
+
+@irdl_op_definition
+class StreamingRegionOp(IRDLOperation):
+    name = "snax_stream.streaming_region"
+
+    # inputs and outputs are pointers to the base address of the data in memory
+    inputs = var_operand_def(IndexType)
+    outputs = var_operand_def(IndexType)
+
+    # streaming stride pattern
+    # there should be one stride pattern for every input/output
+    # the upper bounds of all stride patterns should be equal
```

Comment:
Can you add verifiers for this? Also, do all strides need to be positive?

Reply:
If it's all right I will do this later. The next step is to assign every snax_stream an accelerator which must have the streamer interface implemented. This interface has all the information how the streamers look like and gives us a much better information for verification of the stride pattern.

Reply:
Okay to do later, but not sure I agree with your verification plan.
There's a bit of verification you can already do without lowering to a specific accelerator.
If you insert an incorrect operation of this kind at some point in a pass, you would want the verifier to already report this before lowering. The things you describe in the comments above should be possible to check without any accelerator information, no?

Reply:
Yes but this op can only be lowered to if there is an accelerator with streamers in the system. I think I'll add the operatator accfg definition as an operand of this op.